src/main/java/Duke.java

```diff
@@ -1,10 +1,251 @@
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(ArrayList<Task> tasks) {
+        for (int i = 0; i < tasks.size(); i++) {
+            StringBuilder sb = new StringBuilder();
+            String done = tasks.get(i).isPending() ? "[ ]" : "[X]";
+            sb.append(i + 1).append(": ").append(tasks.get(i).getType())
+                    .append(done).append(" ").append(tasks.get(i).getTaskDesc());
+            if (!tasks.get(i).getAdditionalInfo().isEmpty()) {
+                sb.append(" ").append(tasks.get(i).getAdditionalInfo());
+            }
+            System.out.println(sb);
+        }
+    }
+
+    private static int extractIndexToModify(String[] inputWords) {
+
+        int num = -1;
+        try {
+            num = Integer.parseInt(inputWords[1]) - 1;
+        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+            System.out.println(SOMETHING_WRONG
+                    + "Please input the item number you wish to edit.");
+        }
+        return num;
+
+    }
+
+    private static void deleteTask(String[] inputWords, ArrayList<Task> tasks) {
+
+        for (int i = inputWords.length - 1; i > 0; i--) {
+            try {
+                int num = Integer.parseInt(inputWords[i]) - 1;
+                System.out.println("Task " + (num + 1) + ": "
+                        + tasks.get(num).getTaskDesc() + " has been deleted");
+                if (tasks.size() == 1) {
+                    tasks.clear();
+                } else {
+                    tasks.remove(num);
+                }
+                nextAdd--;
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println(e);
+                System.out.println(SOMETHING_WRONG +
+                        "Please ensure that you have entered the index of the task(s) to be marked as done");
+                break;
+            }
+        }
+
+    }
+
+    private static void tasksCRUD(String input, ArrayList<Task> tasks) {
+
+        String[] inputWords = input.split(" ");
+        if (inputWords[0].equals("done")) {
+            makeTaskDone(inputWords, tasks);
+        } else if (inputWords[0].equals("delete")) {
+            deleteTask(inputWords, tasks);
+        } else {
+//            if (inputWords.length < 2) {
+//                System.out.println(SOMETHING_WRONG + "Please enter more information after your command.");
+//            }
+            if (inputWords[0].equals("deadline")) {
+                int num = extractIndexToModify(inputWords);
+                if (num >= 0) {
+                    Deadline deadline = new Deadline(tasks.get(num).getTaskDesc());
+                    String deadlineString = createRemainingString(inputWords);
+                    deadline.setByDateTime(deadlineString);
+                    tasks.set(num, deadline);
+                    System.out.println("Deadline set: " + deadlineString);
```

Comment:
this chunk of coding is similar to 127 to 131 one, perhaps improve abstraction and readability by have one method for each (deadline, event...)